src/components/TreeView.js

```diff
@@ -56,9 +59,44 @@ function RowValue(props) {
     separator = ': ';
   }
 
+  if (position === 'TransactionEnvelope.signatures') {
+    checkSignatures({signatures: node, fetchedSigners});
+    return formatSignatureCheckState({node, separator})
+  }
+
+  if (position.match(/^TransactionEnvelope\.signatures\[[0-9]+\]\.signature$/)) {
+    return formatSignature({node, separator});
+  }
+
   return <span><strong>{node.type}</strong>{separator}{value}</span>
 }
 
+function checkSignatures({signatures, fetchedSigners}) {
```

Comment:
Any reason for destructuring here and other methods? I think it's just copied from `convertTypedValue({type, value})` definition where it's helpful because you can pass a node value there: `convertTypedValue(node.value)`.